### PR TITLE
WinRM/PSRP: Ensure shell returns UTF-8 output

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR makes UTF-8 output work in PSRP shells.

This relates to #46998

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
WinRM/PSRP

##### ANSIBLE VERSION
v2.8

### CODE REVIEW

**Comment by jborean93** [comment]
Local tests seem to be fine, now we have a psrp only testing location are you able to add a win_shell or win_command test with unicode characters to https://github.com/ansible/ansible/blob/devel/test/integration/targets/connection_psrp/tests.yml. This will test that we don't loose the unicode chars end to end.

File: test/integration/targets/connection_psrp/tests.yml [hunk 1]
```diff
@@ -59,3 +59,28 @@
...
+  assert:
+    that:
+    - command_unicode_output is changed
+    - command_unicode_output.rc == 0
+    - command_unicode_output.stdout == '💩'
+    - command_unicode_output.stderr == ''
+
+- name: Output unicode characters from Powershell using PSRP
+  win_shell: Write-Host '💩'
+  connection: psrp
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: No need for this here

File: changelogs/fragments/psrp-utf8-stdio.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- psrp - Fix UTF-8 output - https://github.com/ansible/ansible/pull/46998
+
```
Comment (jborean93) [inline changelogs/fragments/psrp-utf8-stdio.yaml]: Remove this newline

**Review by nitzmahone — APPROVED** [review]

File: test/integration/targets/connection_psrp/tests.yml [hunk 3]
```diff
@@ -59,3 +59,42 @@
     assert:
       that:
       - async_out.stdout_lines == ["abc"]
+
+- name: Output unicode characters from Powershell using PSRP
+  raw: |
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: Not needed, this is tested above

File: test/integration/targets/connection_psrp/tests.yml [hunk 4]
```diff
@@ -59,3 +59,42 @@
...
+- name: Assert unicode output
+  assert:
+    that:
+    - raw_unicode_output is changed
+    - raw_unicode_output.rc == 0
+    - raw_unicode_output.stdout == '💩'
+    - raw_unicode_output.stderr == ''
+
+- name: Output unicode characters from Powershell using PSRP
+  win_command: |
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: do `win_command: "powershell.exe -ExecutionPolicy ByPass -Command \"Write-Host '\U0001F4A9'\""`. Due to the PyYAML issue you can't put unicode characters that are a surrogate pair in YAML files.

File: test/integration/targets/connection_psrp/tests.yml [hunk 5]
```diff
@@ -59,3 +59,42 @@
...
+    powershell.exe -ExecutionPolicy ByPass -Command "Write-Host '💩'"
+  register: command_unicode_output
+
+- name: Assert unicode output
+  assert:
+    that:
+    - command_unicode_output is changed
+    - command_unicode_output.rc == 0
+    - command_unicode_output.stdout == '💩'
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: `"command_unicode_output.stdout == '\U0001F4A9\n'"`. Need to use the `\U` form and add `\n` to the end.

File: test/integration/targets/connection_psrp/tests.yml [hunk 6]
```diff
@@ -59,3 +59,42 @@
...
+  win_shell: |
+    Write-Host '💩'
+  register: shell_unicode_output
+
+- name: Assert unicode output
+  assert:
+    that:
+    - shell_unicode_output is changed
+    - shell_unicode_output.rc == 0
+    - shell_unicode_output.stdout == '💩'
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: `"shell_unicode_output.stdout == '\U0001F4A9\n'"`. Need to use the `\U` form and add `\n` to the end.

File: test/integration/targets/connection_psrp/tests.yml [hunk 7]
```diff
@@ -59,3 +59,42 @@
...
+- name: Assert unicode output
+  assert:
+    that:
+    - command_unicode_output is changed
+    - command_unicode_output.rc == 0
+    - command_unicode_output.stdout == '💩'
+    - command_unicode_output.stderr == ''
+
+- name: Output unicode characters from Powershell using PSRP
+  win_shell: |
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: do `win_shell: "Write-Host '\U0001F4A9'"`

**Review by jborean93 — CHANGES_REQUESTED** [review]

**Comment by jborean93** [comment]
@dagwieers you also need to indent the new tests to it lines up with the other tasks.

**Review by jborean93 — APPROVED** [review]

File: test/integration/targets/connection_psrp/tests.yml [hunk 8]
```diff
@@ -59,3 +59,27 @@
...
+  - name: Output unicode characters from Powershell using PSRP
+    win_command: powershell.exe -ExecutionPolicy ByPass -Command "Write-Host '\U0001F4A9'"
+    register: command_unicode_output
+
+  - name: Assert unicode output
+    assert:
+      that:
+      - command_unicode_output is changed
+      - command_unicode_output.rc == 0
+      - command_unicode_output.stdout == "\U0001F4A9\n"
```
Comment (jborean93) [inline test/integration/targets/connection_psrp/tests.yml]: Hopefully this is the last one, you need to quote this entire line for the `\U` to take effect, e.g. `"command_unicode_output.stdout == '\U0001F4A9\n'"`. Same as the win_shell assertion below.

**Comment by jborean93** [comment]
@dagwieers I've cherry-picked this commit to https://github.com/ansible/ansible/pull/47530 so they will be merged together.

**Comment by dagwieers** [comment]
@jborean93 Thanks !

**Pull request merged**